invenio_global_search/ext.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# invenio-global-search is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Global Search for InvenioRDM."""
+
+
+from . import config
+
+
+class InvenioGlobalSearch:
+    """InvenioGlobalSearch."""
+
+    def __init__(self, app):
+        """Construct InvenioGlobalSearch."""
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+        """Flask application initialization."""
+        self.init_config(app)
+        app.extensions["invenio-global-search"] = self
+
+    def init_config(self, app):
+        """Initialize configuration."""
+        for k in dir(config):
+            if k.startswith("GLOBAL_SEARCH_"):
+                app.config.setdefault(k, getattr(config, k))
+            elif k.startswith("DUBLIN_CORE_"):
+                app.config.setdefault(k, getattr(config, k))
```

Comment:
This might lead to unforeseen issues. A module should probably only initialize its own values.
It can be specified at an instance specific config module, `app-rdm` or `invenio.cfg`.

Reply:
yes that is true, but `app-rdm` is not possible but `invenio.cfg`is. my problem with the approach to put everything into `invenio.cfg` is that to use the `global-search` i have to put a lot of configuration into `invenio.cfg`.